Fixes #37063 - Add feature in katello-certs-check to verify if CA bundle has any certificates with trust rules

jpasqualetto opened the pull request.

ehelms commented on bin/katello-certs-check:
```diff
@@ -182,6 +182,18 @@ function check-ca-bundle-size () {
+    CHECK=$(awk -v cmd='openssl x509 -noout -text' ' /BEGIN/{close(cmd)};{print | cmd}'  $CA_BUNDLE_FILE |grep 'Trusted Uses'|wc -l)
+    printf $CHECK
+    if [[ $CHECK -lt 1 ]]; then
+                success
```
Can you fix the indentation here to align with the rest of the file?

jpasqualetto replied:
I don't see what's wrong with indentation here. What's the problematic line?

ehelms replied:
The file has 4 space indentation and this all lines should be consistent with it.

jpasqualetto replied:
Ack. Thanks for clarifying.

ehelms commented on bin/katello-certs-check:
```diff
@@ -182,6 +182,18 @@ function check-ca-bundle-size () {
+                success
+    else
+            error 10 "The CA bundle contains $CHECK certificate(s) with trust rules. This may create problems for older systems to trust the bundle. Please, recreate the bundle using certificates without trust rules"
+                echo
```
Can you fix the indentation here to align with the rest of the file?

jpasqualetto replied:
sure, will do

ehelms commented:
Do you think you could add a test case for this? We have certificate fixtures where we generate different of these failing scenarios (https://github.com/theforeman/foreman-installer/blob/develop/spec/fixtures/katello-certs-check/create_cert.sh) and then we test that the script does what we expect when encountering said type (https://github.com/theforeman/foreman-installer/blob/develop/spec/katello_certs_check_spec.rb).

jpasqualetto commented:
I'll have a look at the test case. I may need some help with it.

jpasqualetto commented:
> I'll have a look at the test case. I may need some help with it.

I pushed an update with a test case and fixed the indentation issues.  Let me know if anything else is needed.

ehelms commented on spec/fixtures/katello-certs-check/create_cert.sh:
```diff
@@ -28,6 +28,16 @@ else
 fi
 
+CA_BUNDLE=ca-bundle-trust-rules
+CA_CERT_WITH_TRUST_RULES=ca_with_trust_rules
```
Could this follow the same name convention of using dashes?

jpasqualetto replied:
Sure. Updated

ehelms commented on bin/katello-certs-check:
```diff
@@ -182,6 +182,18 @@ function check-ca-bundle-size () {
+    CHECK=$(awk -v cmd='openssl x509 -noout -text' ' /BEGIN/{close(cmd)};{print | cmd}'  $CA_BUNDLE_FILE |grep 'Trusted Uses'|wc -l)
+    printf $CHECK
+    if [[ $CHECK -lt 1 ]]; then
+		success
```
```suggestion
	success
```

jpasqualetto replied:
I finally understood what you mean about the indentation. I was copying the indentation of the function check-ca-bundle-size which was wrong.  I was confused because my indentation was matching that one which I assumed was a good template to follow. Guess I was wrong.  I'll adjust the indentation of that one too in my next update.

ehelms commented on bin/katello-certs-check:
```diff
@@ -182,6 +182,18 @@ function check-ca-bundle-size () {
+    if [[ $CHECK -lt 1 ]]; then
+		success
+    else
+		error 10 "The CA bundle contains $CHECK certificate(s) with trust rules. This may create problems for older systems to trust the bundle. Please, recreate the bundle using certificates without trust rules"
```
```suggestion
	error 10 "The CA bundle contains $CHECK certificate(s) with trust rules. This may create problems for older systems to trust the bundle. Please, recreate the bundle using certificates without trust rules"
```

jpasqualetto replied:
ack

ehelms commented on bin/katello-certs-check:
```diff
@@ -182,6 +182,18 @@ function check-ca-bundle-size () {
+		success
+    else
+		error 10 "The CA bundle contains $CHECK certificate(s) with trust rules. This may create problems for older systems to trust the bundle. Please, recreate the bundle using certificates without trust rules"
+		echo
```
```suggestion
	echo
```

jpasqualetto committed:
Fixes #37063 - Add feature in katello-certs-check to verify if CA bundle has any certificates with trust rules

 - Add feature to verify if provided CA bundle has certificates with trust rules
 - Add test case for the new feature
 - Fix indentation on function check-ca-bundle-size
